analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -84,7 +87,8 @@ public abstract class UtilityAnalyzerBase<TMessage> : UtilityAnalyzerBase
             context.RegisterCompilationAction(c =>
                 {
                     ReadParameters(context, c);
-                    if (!IsAnalyzerEnabled)
+                    if (!IsAnalyzerEnabled ||
+                        (SkipAnalysisForTestProject && IsTestProject))
```

Comment:
Shouldn't we write an empty file to avoid the warning on the plugin side in this case? To declare we've met this stage and intentionally decided not to produce the data.